### PR TITLE
 feat(dca): Position detail

### PR DESCRIPTION
**Pull request by th0rOdinson**
Screenshot 2024-03-14 at 15 09 30

### CODE REVIEW

File: apps/root/src/pages/position-detail/frame/index.tsx [hunk 1]
```diff
@@ -117,460 +81,76 @@ const PositionDetailFrame = () => {
...
-            {useProtocolToken && !hasPermission && hasSignSupport && (
-              <Typography variant="body">
-                <FormattedMessage
-                  description="Approve signature companion text"
-                  defaultMessage="You will need to first sign a message (which is costless) to authorize our Companion contract. Then, you will need to submit the transaction where you get your balance back as {from}."
-                  values={{ from: position.from.symbol }}
-                />
+          <StyledTab
+            disableRipple
+            sx={{ marginLeft: '32px' }}
```
Comment (FiboApe) [inline apps/root/src/pages/position-detail/frame/index.tsx:127]: Gotcha! no px :D

File: apps/root/src/pages/position-detail/components/summary-container/index.tsx [hunk 2]
```diff
@@ -32,50 +28,54 @@ const StyledFlexGridItem = styled(Grid)`
...
+        <Sticky enabled={!isDownMd} top={95}>
+          <StyledPaper>
+            {isLoading ? (
+              <PositionDataSkeleton />
+            ) : (
               <Details
                 position={position}
                 pendingTransaction={pendingTransaction}
                 yieldOptions={yieldOptions}
-                totalGasSaved={totalGasSaved}
```
Comment (FiboApe) [inline apps/root/src/pages/position-detail/components/summary-container/index.tsx:58]: I think we should still show this for eth mainnet

File: apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx [hunk 3]
```diff
@@ -240,36 +85,41 @@ const Details = ({ position, pendingTransaction, yieldOptions, totalGasSaved }:
...
-  const toFullPrice = parseUsdPrice(position.to, swappedBase, toUsdPrice);
-  const toYieldFullPrice = parseUsdPrice(position.to, swappedYield?.amount, toUsdPrice);
-  const showToFullPrice = !!toFullPrice;
-  const showToYieldFullPrice = !!toYieldFullPrice;
-  const showToPrice = !!toPrice;
-  const showToYieldPrice = !!toYieldPrice;
-  const showRatePrice = !!ratePrice;
-  const showFromPrice = !!fromPrice;
-  const showFromYieldPrice = !!fromYieldPrice;
+  const toWithdrawPrice = parseUsdPrice(position.to, toWithdraw.amount, toUsdPrice);
```
Comment (FiboApe) [inline apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx:119]: The prices should actually already be inside the token :)

File: apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx [hunk 4]
```diff
@@ -279,465 +129,376 @@ const Details = ({ position, pendingTransaction, yieldOptions, totalGasSaved }:
...
-
   const executedSwaps = Number(totalSwaps) - Number(remainingSwaps);
 
   const isOldVersion = !VERSIONS_ALLOWED_MODIFY.includes(position.version);
 
   const isTestnet = TESTNETS.includes(positionNetwork.chainId);
 
+  const { mainCurrencyToken } = getNetworkCurrencyTokens(positionNetwork);
+
+  const positionStatusLabel = React.useMemo(() => {
```
Comment (FiboApe) [inline apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx:140]: We could probably have this as a separate component right? so we declutter a bit the code from this one

File: apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx [hunk 5]
```diff
@@ -279,465 +129,376 @@ const Details = ({ position, pendingTransaction, yieldOptions, totalGasSaved }:
...
               {averageBuyPrice > 0n ? (
                 <FormattedMessage
                   description="positionDetailsAverageBuyPrice"
                   defaultMessage="1 {from} = {currencySymbol}{average} {to}"
                   values={{
-                    b: (chunks: React.ReactNode) => <b>{chunks}</b>,
                     from: tokenFromAverage.symbol,
-                    to: STABLE_COINS.includes(tokenToAverage.symbol) ? 'USD' : tokenToAverage.symbol,
-                    average: formatCurrencyAmount(averageBuyPrice, tokenToAverage, 4),
+                    to: !STABLE_COINS.includes(tokenToAverage.symbol) ? tokenToAverage.symbol : '',
```
Comment (FiboApe) [inline apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx:357]: should we have at least the `$` symbol if it is a stablecoin?

File: apps/root/src/pages/position-detail/components/summary-container/index.tsx (hunk 2 above)
Comment (th0rOdinson) [reply apps/root/src/pages/position-detail/components/summary-container/index.tsx:58]: This has been moved to position-data directly

File: apps/root/src/common/components/composed-token-icon/index.tsx [hunk 6]
```diff
@@ -43,9 +44,25 @@ interface TokenButtonProps {
   withNetwork?: boolean;
   isInChip?: boolean;
   size?: number;
+  isLoading?: boolean;
 }
 
-const ComposedTokenIcon = ({ tokenTop, tokenBottom, isInChip, size, withNetwork }: TokenButtonProps) => {
+const ComposedTokenIcon = ({ tokenTop, tokenBottom, isInChip, size, withNetwork, isLoading }: TokenButtonProps) => {
+  const theme = useTheme();
+  if (isLoading && size) {
```
[the remark this answers is not in the document]
Comment (th0rOdinson) [reply apps/root/src/common/components/composed-token-icon/index.tsx:52]: I was using it to calculate px conversion, in order to pass it to the Skeleton. But having a default value, as we do in TokenIcon, works better

File: apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx (hunk 3 above)
Comment (th0rOdinson) [reply apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx:119]: Oh nice, forgot about this one!

File: apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx (hunk 5 above)
Comment (th0rOdinson) [reply apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx:357]: It does! I kept the 'currencySymbol' value

File: apps/root/src/pages/position-detail/frame/index.tsx (hunk 1 above)
Comment (th0rOdinson) [reply apps/root/src/pages/position-detail/frame/index.tsx:127]: I took the time to apply styles to the whole frame based on this comment 😄 . We didn't have a task so now it's included here

File: apps/root/src/pages/dca/create-position/components/step2/components/yield-selector/index.tsx [hunk 7]
```diff
@@ -84,11 +84,21 @@ const YieldSelector = ({
...
+    } else {
+      dispatch(setFromYield(null));
+    }
     trackEvent('DCA - Set yield from', {});
   };
   const onSetToYield = (newYield: YieldOption | null) => {
-    dispatch(setToYield(newYield));
+    if (newYield) {
+      const { apy, name, token, tokenAddress } = newYield;
+      dispatch(setToYield({ apy, name, token, tokenAddress }));
```
Comment (FiboApe) [inline apps/root/src/pages/dca/create-position/components/step2/components/yield-selector/index.tsx:98]: so why not just (setToYield(newYield))?

**Review by FiboApe — APPROVED** [review]

**Pull request merged**